### PR TITLE
Fix decoding unnamed params, the return type was right, but names weren't populated for the arg itself.

### PR DESCRIPTION
**Pull request by nolag**

### CODE REVIEW

**Review by cedric-cordenier — DISMISSED** [review]

File: core/services/relay/evm/types/codec_entry.go [hunk 1]
```diff
@@ -146,11 +146,13 @@ func (entry *codecEntry) Init() (err error) {
 			}
 			for {
 				name = name + "_X"
+				args[i].Name = name
 				if !seenNames[name] {
 					break
 				}
 			}
 		}
+		args[i] = arg
```
Comment (reductionista) [inline core/services/relay/evm/types/codec_entry.go:155]: Doesn't this undo the earlier `args[i].Name = name` assignment in the for loop?

File: core/services/relay/evm/types/codec_entry.go [hunk 2]
```diff
@@ -146,11 +146,13 @@ func (entry *codecEntry) Init() (err error) {
 			}
 			for {
 				name = name + "_X"
+				args[i].Name = name
```
Comment (reductionista) [inline core/services/relay/evm/types/codec_entry.go]: Maybe you want `arg.Name = name` instead?

File: core/services/relay/evm/types/codec_entry.go (hunk 1 above)
Comment (reductionista) [reply core/services/relay/evm/types/codec_entry.go:155]: I think you meant to set `entry.args[i] = arg` here, right?

Comment (nolag) [reply core/services/relay/evm/types/codec_entry.go:155]: It's always the same slice in this case.  Unless there's a single unnamed arg, it's the same slice.

Comment (nolag) [reply core/services/relay/evm/types/codec_entry.go:155]: Fixed the name.

**Review by cedric-cordenier — APPROVED** [review]

File: core/services/relay/evm/types/codec_entry.go (hunk 1 above)
Comment (reductionista) [reply core/services/relay/evm/types/codec_entry.go:155]: Oh, I think I see... so, unwrapNames only creates a new slice for `args` if it's a single unnamed arg, but the name only needs to be updated if there are multiple args right? Seems potentially brittle, but I guess that works.

**Review by reductionista — APPROVED** [review]

**Pull request merged**